arithmetization/src/test/java/net/consensys/linea/zktracer/instructionprocessing/callTests/sixtyThreeSixtyFourths/SixtyThreeSixtyFourthsTests.java

```diff
@@ -0,0 +1,390 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class SixtyThreeSixtyFourthsTests {
+
+  /*
+  Cases to cover:
+
+  * value = 0
+  If precompileGasCost >= 2300 then we are interested in:
+   * value = 1, targetAddressExists = false
+   * value = 1, targetAddressExists = true
+
+  Note: BLAKE2F and MODEXP requires a non-zero non-trivial input to have a cost greater than 2300.
+  Other precompiles only require a proper call data size at most.
+  */
+
+  final Bytes INFINITE_GAS = Bytes.fromHexString("ff".repeat(32));
+
+  // BLAKE2F specific parameters
+  final int rLeadingByte = 0x09;
+  final int r = rLeadingByte << 8;
+
+  // MODEXP specific parameters
+  int bbs = 2;
+  int ebs = 6;
+  int mbs = 128;
+  int exponentLog = 0;
+  List<ToyAccount> additionalAccounts = new ArrayList<>();
+  Address codeOwnerAddress = Address.fromHexString("0xC0DE");
+
+  // Cost of preCallProgram in different scenarios:
+  // (address, transfersValue) -> gasCost
+  final Map<Address, Map<Boolean, Long>> preCallProgramGasMap =
+      Stream.of(
+              ECREC,
+              SHA256,
+              RIPEMD160,
+              ID,
+              MODEXP,
+              ALTBN128_ADD,
+              ALTBN128_MUL,
+              ALTBN128_PAIRING,
+              BLAKE2B_F_COMPRESSION)
+          .collect(
+              Collectors.toMap(
+                  address -> address,
+                  address ->
+                      new HashMap<>() {
+                        {
+                          put(
+                              false,
+                              BytecodeRunner.of(preCallProgram(address, false, false, 0))
+                                  .runOnlyForGasCost(additionalAccounts));
+                          put(
+                              true,
+                              BytecodeRunner.of(preCallProgram(address, false, true, 0))
+                                  .runOnlyForGasCost(additionalAccounts));
+                        }
+                      }));
+
+  // Note: transferValue = false and cds = 0 as we are interested only in the cost of the
+  // corresponding PUSHes here
+
+  @ParameterizedTest
+  @MethodSource("fixedCostLTStipendTestSource")
+  void fixedCostLTStipendTest(long gasLimit, boolean insufficientGasForPrecompileExpected) {
+    // Only ECADD falls in this scenario and whenever transferValue = true, gas is enough
+    // Thus, we only test the case in which transferValue = false
+
+    final BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    program.immediate(preCallProgram(ALTBN128_ADD, false, false, 0)).op(CALL);
+
+    final BytecodeRunner bytecodeRunner = BytecodeRunner.of(program);
+    bytecodeRunner.run(gasLimit);
+
+    // insufficientGasForPrecompileExpected = true  =>
+    // targetCalleeGas = 63/64 * (252 - 100) = 150
+    // insufficientGasForPrecompileExpected = false =>
+    // targetCalleeGas = 63/64 * (251 - 100) = 149
```

Comment:
Can this comment be deleted ? The `(63/64) * bla` stuff in particular is ambiguous (and despite the common name of **(63/64)ths** the gas calculation does not involve any such multiplication, in any order. If anything it's more about doing **1 - (1/64)ths**.